File: src/count_divs.py
import re

with open('resources/views/livewire/user/profile.blade.php', 'r') as f:
    content = f.read()

# Strip HTML comments
content = re.sub(r'<!--.*?-->', '', content, flags=re.DOTALL)
# Strip blade comments
content = re.sub(r'\{\{--.*?--\}\}', '', content, flags=re.DOTALL)

lines = content.split('\n')
depth = 0
for i, line in enumerate(lines):
    # simple lowercasing and counting
    lower_line = line.lower()
    opens = lower_line.count('<div')
    closes = lower_line.count('</div')
    depth += opens - closes
    
    if depth < 0:
        print(f"NEGATIVE DEPTH at line {i+1}: {line}")
        depth = 0 # reset to find next
        
print(f"Final depth: {depth}")

File: src/count_divs.js
const fs = require('fs');
let content = fs.readFileSync('resources/views/livewire/user/profile.blade.php', 'utf8');

// Strip html comments
content = content.replace(/<!--[\s\S]*?-->/g, '');
// Strip blade comments
content = content.replace(/{{--[\s\S]*?--}}/g, '');

const lines = content.split('\n');
let depth = 0;
for(let i=0; i<lines.length; i++) {
    const opens = (lines[i].match(/<div(\s|>)/gi) || []).length;
    const closes = (lines[i].match(/<\/div>/gi) || []).length;
    depth += opens - closes;
    if (opens !== closes) {
        // console.log(`${i+1}: ${depth} (+${opens} -${closes})`);
    }
    if (depth < 0) {
        console.log(`NEGATIVE DEPTH at line ${i+1}: ${lines[i]}`);
    }
}
console.log(`Final depth: ${depth}`);
